### PR TITLE
fix: add hostname to span during stress relief mode

### LINKED ISSUE #641: Stress Relief doesn't respect AddHostMetadataToTrace
If Stress Relief activates, it doesn't decorate the trace with meta.refinery.local_hostname.

It should; it's useful in diagnostics.

We should also consider if there are other meta fields that would also be valuable.

### PR DESCRIPTION
**Pull request by fchikwekwe**
Closes #641 

Add hostname to spans where stress relief mode is activated.

### CODE REVIEW

File: collect/collect.go [hunk 1]
```diff
@@ -707,6 +712,9 @@ func (i *InMemCollector) send(trace *types.Trace, reason string) {
 		if i.Config.GetAddRuleReasonToTrace() {
 			sp.Data["meta.refinery.reason"] = reason
 		}
+		if i.Config.GetStressReliefConfig().Mode != "" {
+			sp.Event.Data["meta.refinery.local_hostname"] = i.hostname
```
Comment (TylerHelmuth) [inline collect/collect.go:716]: It looks like this function is already setting `local_hostname` whenever `i.hostname` is set on line 731.

File: collect/collect.go [hunk 2]
```diff
@@ -559,7 +559,9 @@ func (i *InMemCollector) ProcessSpanImmediately(sp *types.Span, keep bool, sampl
 	if i.Config.GetAddRuleReasonToTrace() {
 		sp.Event.Data["meta.refinery.reason"] = reason
 	}
-
+	if i.Config.GetStressReliefConfig().Mode != "" {
```
Comment (TylerHelmuth) [inline collect/collect.go:562]: Can we use the same `if i.hostname !=nil` logic that is used in the `send` function?

File: collect/collect.go [hunk 3]
```diff
@@ -571,6 +573,9 @@ func (i *InMemCollector) ProcessSpanImmediately(sp *types.Span, keep bool, sampl
 func (i *InMemCollector) dealWithSentTrace(keep bool, sampleRate uint, spanCount uint, sp *types.Span) {
 	if i.Config.GetAddRuleReasonToTrace() {
 		sp.Data["meta.refinery.reason"] = "late"
+
+	if i.Config.GetStressReliefConfig().Mode != "" {
```
Comment (TylerHelmuth) [inline collect/collect.go:577]: Can we use the same `if i.hostname !=nil` logic that is used in the `send` function?

File: collect/collect.go (hunk 1 above)
Comment (fchikwekwe) [reply collect/collect.go]: Okay, changed all instances to adopt that pattern.

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**